VK_App/VK_App/Controller/FriendsListTableViewController.swift

```diff
@@ -38,20 +50,66 @@ final class FriendsListTableViewController: UITableViewController {
         }
     }
 
+    // MARK: - Private Methods
+
+    private func setupUI() {
+        tableViewConfiguration()
+        createSectionTitle()
+    }
+
+    private func createSectionTitle() {
+        for people in userFriends {
+            guard let firstLatter = people.userName.first else { return }
+            if sections[firstLatter] != nil {
+                sections[firstLatter]?.append(people.userName)
+            } else {
+                sections[firstLatter] = [people.userName]
+            }
+        }
+        sectionsTitle = Array(sections.keys).sorted()
+    }
+
+    private func tableViewConfiguration() {
+        tableView.register(
+            UINib(
+                nibName: Constants.sectionHeaderViewName,
+                bundle: nil
+            ),
+            forHeaderFooterViewReuseIdentifier: Constants.headerForSectionIdentifier
+        )
+        tableView.sectionFooterHeight = 0
+    }
+
     // MARK: - Table view data source
 
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        sections.count
+    }
+
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        userFriends.count
+        sections[sectionsTitle[section]]?.count ?? 0
     }
 
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let user = userFriends[indexPath.row]
+        guard let people = sections[sectionsTitle[indexPath.section]]?[indexPath.row] else { fatalError() }
         guard let cell = tableView.dequeueReusableCell(
             withIdentifier: Constants.cellIdentifier,
             for: indexPath
         ) as? FriendsListViewCell
         else { return UITableViewCell() }
-        cell.setupInfo(user)
+        cell.setupInfo(user, people)
         return cell
     }
+
+    override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let header = tableView
+            .dequeueReusableHeaderFooterView(withIdentifier: Constants.headerForSectionIdentifier) as? SectionHeaderView
+        else {
+            return UIView()
+        }
+        let name = String(sectionsTitle[section])
+        header.sectionNameLabel.text = String(name)
+        return header
+    }
```

Comment:
```suggestion
    } -> String
```
Есть метод возвращающий String вместо View. Если меняешь свойства view (типа backGround), то используем этот. Если достаточно вернуть String, то этот метод более сложный